### PR TITLE
EVG-5928: use jasper for all subprocess operations

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: command/exec.go [hunk 1]
```diff
@@ -235,27 +222,21 @@ func (c *subprocessExec) Execute(ctx context.Context, comm client.Communicator,
...
 	}
 
-	if err := proc.Start(ctx); err != nil {
-		return errors.Wrap(err, "problem starting background process")
-	}
-	pid := proc.GetPid()
-	subprocess.TrackProcess(taskID, pid, logger.System())
-
-	if c.Background {
-		logger.Execution().Infof("started background process with pid %d", pid)
```
Comment (bsamek) [inline command/exec.go]: These log messages are sometimes valuable. Does jasper do something similar?

File: command/shell.go [hunk 2]
```diff
@@ -154,26 +160,9 @@ func (c *shellExec) Execute(ctx context.Context,
...
 	}
 
-	if err = localCmd.Start(ctx); err != nil {
-		logger.System().Debugf("error spawning shell process: %v", err)
-		return err
-	}
-
-	pid := localCmd.GetPid()
-
-	logger.System().Debugf("spawned shell process with pid %d", pid)
```
Comment (bsamek) [inline command/shell.go]: Same question here about logging the pid to the task's logs.

File: command/shell.go [hunk 3]
```diff
@@ -154,26 +160,9 @@ func (c *shellExec) Execute(ctx context.Context,
...
-
-	// Call the platform's process-tracking function. On some OSes this will be a noop,
-	// on others this may need to do some additional work to track the process so that
-	// it can be cleaned up later.
-	subprocess.TrackProcess(conf.Task.Id, pid, logger.System())
```
Comment (bsamek) [inline command/shell.go]: Where do these procs get cleaned up?

File: units/jasper_cleanup.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func (j *jasperManagerCleanup) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	j.env.JasperManager().Clear(ctx)
```
Comment (bsamek) [inline units/jasper_cleanup.go]: Is this safe to call while procs are running?

File: agent/agent.go [hunk 5]
```diff
@@ -77,7 +77,7 @@ func New(opts Options, comm client.Communicator) (*Agent, error) {
 		comm: comm,
 	}
 
-	jpm, err := jasper.NewLocalManager(true)
+	jpm, err := jasper.NewLocalManager(false)
```
[the remark this answers is not in the document]
Comment (tychoish) [reply agent/agent.go]: To avoid situations where we start tasks in nested job objects.

File: command/shell.go (hunk 3 above)
Comment (tychoish) [reply command/shell.go]: We're still calling the legacy TrackProcess code (it's just called in the process constructor.

The cleanup, as usual, is in the agent/inter (and pre) task running.

File: units/jasper_cleanup.go (hunk 4 above)
Comment (tychoish) [reply units/jasper_cleanup.go]: yup, there are locks.

File: command/exec.go (hunk 1 above)
Comment (tychoish) [reply command/exec.go]: All of this logging happens in the process constructor now and should still be around.

File: command/shell.go (hunk 2 above)
Comment (tychoish) [reply command/shell.go]: this happens in the process constructor.

File: model/host/hostutil.go [hunk 6]
```diff
@@ -77,44 +77,13 @@ func (h *Host) RunSSHCommand(ctx context.Context, cmd string, sshOptions []strin
...
-	})
+	cmdArgs := append(append(append([]string{"ssh"}, "-p", hostInfo.Port, "-t", "-t"), sshOptions...), cmd)
 
 	var cancel context.CancelFunc
 	ctx, cancel = context.WithTimeout(ctx, sshTimeout)
 	defer cancel()
 
-	err = proc.Run(ctx)
-	grip.Notice(proc.Stop())
+	err = env.JasperManager().CreateCommand(ctx).ApplyFromOpts(&jasper.CreateOptions{Output: opts}).Add(cmdArgs).Run(ctx)
```
Comment (Kimchelly) [inline model/host/hostutil.go]: How does this command know the remote host if `Host()` isn't called?

**Pull request merged**